src/rpcwallet.cpp

```diff
@@ -745,6 +745,19 @@ UniValue getbalance(const UniValue& params, bool fHelp)
 }
 
 
+
+UniValue getunconfirmedbalance(const UniValue& params, bool fHelp)
+{
+    if (fHelp || params.size() > 0)
+        throw runtime_error("getunconfirmedbalance: returns unconfirmed balance in wallet\n");
+
+    LOCK2(cs_main, pwalletMain->cs_wallet);
```

Comment:
Since `CWallet::GetUnconfirmedBalance()` already manages these locks, should we do it again? I don't think it hurts, but it may be a gotcha down the line...

Reply:
Agreed. Done.